system/src/system_network_manager_api.cpp

```diff
@@ -308,6 +308,21 @@ void network_off(network_handle_t network, uint32_t flags, uint32_t param, void*
     }());
 }
 
+void network_wait_modem_off(network_handle_t network, system_tick_t timeout, void*) {
+    SYSTEM_THREAD_CONTEXT_ASYNC_CALL([&]() {
```

Comment:
This does not make sense. A wait cannot be async. Since this is an internal function, we can skip this whole system thread marshalling.